### ISSUE TITLE
Cannot deserialize Theta EmptyCompactSketch from Java

### ISSUE THREAD

**Issue by Lordshinjo** [apache/incubator-datasketches-cpp#149]
In Java, an empty Theta compact sketch is represented by the singleton `EmptyCompactSketch`, which has a seed hash of 0.

It is represented by the constant byte array `[1, 3, 3, 0, 0, 0x1E, 0, 0]`, as defined [here](https://github.com/apache/incubator-datasketches-java/blob/1.3.0-incubating/src/main/java/org/apache/datasketches/theta/EmptyCompactSketch.java#L38).
Deserializing this byte array from Java returns `EmptyCompactSketch` as expected (snippet using [Ammonite](https://ammonite.io/)):
```
@ import $ivy.`org.apache.datasketches:datasketches-java:1.3.0-incubating`, org.apache.datasketches.theta._, org.apache.datasketches.memory._
import $ivy.$                                                           , org.apache.datasketches.theta._, org.apache.datasketches.memory._

@ val emptySketch = (new UpdateSketchBuilder).build.compact
emptySketch: CompactSketch =
### EmptyCompactSketch SUMMARY:
   Estimate                : 0.0
   Upper Bound, 95% conf   : 0.0
   Lower Bound, 95% conf   : 0.0
   Theta (double)          : 1.0
   Theta (long)            : 9223372036854775807
   Theta (long) hex        : 7fffffffffffffff
   EstMode?                : false
   Empty?                  : true
   Retained Entries        : 0
   Seed Hash               : 0 | 0
### END SKETCH SUMMARY


@ emptySketch.toByteArray
res2: Array[Byte] = Array(1, 3, 3, 0, 0, 30, 0, 0)

@ Sketch.wrap(Memory.wrap(Array[Byte](1, 3, 3, 0, 0, 0x1E, 0, 0)))
res3: Sketch =
### EmptyCompactSketch SUMMARY:
   Estimate                : 0.0
   Upper Bound, 95% conf   : 0.0
   Lower Bound, 95% conf   : 0.0
   Theta (double)          : 1.0
   Theta (long)            : 9223372036854775807
   Theta (long) hex        : 7fffffffffffffff
   EstMode?                : false
   Empty?                  : true
   Retained Entries        : 0
   Seed Hash               : 0 | 0
### END SKETCH SUMMARY
```

In C++ (or Python), the result is a Sketch seed hash mismatch:
```
>>> import datasketches
>>> datasketches.theta_sketch.deserialize(bytes([1, 3, 3, 0, 0, 0x1E, 0, 0]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Sketch seed hash mismatch: expected 37836, actual 0
```
I would expect to get an empty sketch instead.

It looks like an explicit design decision in the Java project to ignore the seed hash for the empty sketch, so I'm raising the issue here instead of over there.

**Comment by leerho** [comment]
Thank you for reporting this!  Are you using code from the formal 1.0.0 Apache release or from a recent pull of master?

**Comment by Lordshinjo** [comment]
I am using code from the current master.

**Comment by jmalkin** [comment]
I'm not yet convinced that Java is handling this correctly.

**Comment by jmalkin** [comment]
The more I think about it, the more I question why Java would ignore the seed on an empty sketch. There might be no hashes, but a mismatch should still be taken a strong signal that things were not properly/consistently configured and are not compatible.

**Comment by leerho** [comment]
We actually made this decision some time ago.  We return empty sketches in many situations where we could have returned a null.  We do this instead of returning nulls because nulls propagate.  Now the user has the option of ignoring the empty sketch as if it were null or not.

Not including the seed hash means we can create a static singleton for the empty sketch, which is very fast.  

Not detecting a seed mismatch on empties (essentially nulls) is not a huge issue.  As soon as the first sketch that comes in with real data the mismatch will be detected.  I can't imagine a dataset of all empties (or nulls) would be very interesting or important as it conveys no information.

**Comment by leerho** [comment]
Due to a number of complex factors we have decided to fix this in C++.   This issue will be closed once the fix is completed.

**Comment by AlexanderSaydakov** [comment]
should be fixed by #151
